comp/api/api/apiimpl/api_test.go

```diff
@@ -164,154 +140,3 @@ func TestStartBothServersWithObservability(t *testing.T) {
 		})
 	}
 }
-
-type s struct {
-	body string
-}
-
-func (s *s) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
-	w.WriteHeader(http.StatusOK)
-	w.Write([]byte(s.body))
-}
-
-type grpcServer struct {
-	grpcServer bool
-	gateway    bool
-}
-
-func (grpc *grpcServer) BuildServer() http.Handler {
-	if grpc.grpcServer {
-		return &s{
-			body: "GRPC SERVER OK",
-		}
-	}
-	return nil
-}
-
-func (grpc *grpcServer) BuildGatewayMux(string) (http.Handler, error) {
-	if grpc.gateway {
-		return &s{
-			body: "GRPC GATEWAY OK",
-		}, nil
-	}
-
-	return nil, fmt.Errorf("error")
-}
-
-func TestStartServerWithGrpcServer(t *testing.T) {
```

Comment:
This test still relevant as we want to test that the gRPC component is being use when building the API component. I would say that we could remove the `gateway` attribute of the `grpcServer` struct and keep the test.